ethdb/privateapi/ethbackend.go

```diff
@@ -193,14 +193,12 @@ func (s *EthBackendServer) EngineExecutePayloadV1(ctx context.Context, req *type
 		return nil, fmt.Errorf("not a proof-of-stake chain")
 	}
 	// Check mandatory fields
-	if req == nil || req.ParentHash == nil || req.BlockHash == nil || req.Coinbase == nil || req.ExtraData == nil ||
-		req.LogsBloom == nil || req.ReceiptRoot == nil || req.StateRoot == nil || req.Random == nil ||
-		req.Transactions == nil {
-
+	if req == nil || req.ParentHash == nil || req.BlockHash == nil || req.Coinbase == nil ||
+		req.ReceiptRoot == nil || req.StateRoot == nil {
```

Comment:
Why are ExtraData, LogsBloom, Random, and Transactions no longer mandatory?

Reply:
because if they are set to 0 which is possible they would be nil, so the check could be wrong in some cases.

Reply:
But they are mandatory according to the Engine API, aren't they?

Reply:
but you can have, say random at 0 or transactions as an empty list, however golang says that `common.Hash{}` and `[][]byte` are also equivalent to `nil`, i think that we have the required flag in gointerfaces anyway though

Reply:
so really, those becomes sanity checks anyway rather than `is mandatory?`

Reply:
OK (though Random should not be set to 0). Perhaps we can remove the checks here altogether as we can rely on `gencodec:"required"`.

Reply:
ok fair enough

Reply:
Random is set to 0 in the kintsugi test vectors, in one of the sample requests